components/brave_new_tab_ui/api/brave_news/news.ts

```diff
@@ -92,16 +92,23 @@ class BraveNewsApi {
   async setChannelSubscribed (channelId: string, subscribed: boolean) {
     // While we're waiting for the new channels to come back, speculatively
     // update them, so the UI has instant feedback.
+    let subscribedLocales = this.lastChannels[channelId]?.subscribedLocales ?? []
+    if (subscribedLocales.includes(channelId)) {
```

Comment:
Are we incorrectly comparing a channel name and locale string here?

Actually this code block is confusing me more than it probably should! Is the purpose to merge a potential existing channel subscription with the same channel id but different locales with the current locale?

Reply:
Ooh nice catch! This was meant to make sure the feedback was instant. I.e. if you're unsubcribing, update the data in the service so this channel is unsubscribed in the current locale. (or if you're subscribing). It gets overridden when the service responds anyway.